app/models/setting/templates.rb

```diff
@@ -0,0 +1,29 @@
+class Setting::Templates < Setting
+  def self.load_defaults
+    return unless super
+
+    %w(templates_filter templates_branch).each { |s| Setting::BLANK_ATTRS << s }
+
+    self.transaction do
+      [
+        self.set('templates_verbose', N_('Choose verbosity for Rake task importing templates'), false, N_('Verbosity')),
+        self.set('templates_associate', N_('Associate templates to OS ("always", "new", "never")'), "new", N_('Associate')),
+        self.set('templates_prefix', N_('The string all imported templates should begin with'), "Community ", N_('Prefix')),
+        self.set('templates_dirname', N_('The directory within the git tree containing the templates'), '/', N_('Dirname')),
+        self.set('templates_filter', N_('Import names matching this regex (case-insensitive; snippets are not filtered)'), nil, N_('Filter')),
+        self.set('templates_repo', N_('Default Git repo to sync from'), 'https://github.com/theforeman/community-templates.git', N_('Repo')),
+        self.set('templates_negate', N_('Negate the search'), false, N_('Negate')),
+        self.set('templates_branch', N_('Default branch in Git repo'), nil, N_('Branch'))
```

Comment:
Yes, it is intentional. If you keep `false`, the in-place form for editing settings assumes the value should be boolean and offers Yes/No alternatives instead of input field.